### DISCUSSION TITLE
Getting all twin information out of orientation.map function

### DISCUSSION THREAD

**Discussion by s162555** [mtex-toolbox/mtex#1803] [Ask Anything]
Hi developers,

I am currently working with Tijmen Vermeij on identifying individual twin systems within zinc coatings and have a question regarding the `orientation.map` function.

If I for instance use the following command:
```matlab
twinning = orientation.map(Miller(0,1,-1,-1,CSlist{2}),Miller(-1,1,0,-1,CSlist{2}),Miller(1,0,-1,1,CSlist{2},'uvw'),Miller(1,0,-1,-1,CSlist{2},'uvw'));
```
I got the following expected output:
```
twinning = misorientation (Zinc -> Zinc)
 
  Bunge Euler angles in degree
  phi1     Phi    phi2    Inv.
   120 93.9656     240       0
```
   The output however takes the symmetry conditions of the HCP crystal into account resulting in one misorientation relation, but is it also possible to get out 6 options by not taking into account the symmetry? This will make it possible to identify the individual twin systems instead of just identifying a twin family. 

   Next thing to do: if we have the 6 contributions for each individual twinning system we use the following command to specifically identify which twin system is active in the analyzed Parent and Child grain:
```matlab
   % restrict to twinnings with threshold 5 degree
isTwinning = angle(gB_ZnZn.misorientation,**twinning(1:6)**) < 5*degree;
twinBoundary = gB_ZnZn(isTwinning)
```
Thanks in advance

**Comment by ralfHielscher** [comment]
You can use the command ```angle``` with the option `'noSymmetry'` or `'noSym1'`, `'noSym2'` to ignore symmetry.

Ralf.

**Comment by s162555** [comment]
Thanks for fast response, but if I understood you correctly you mean the following command for angle:
```matlab
twinning = orientation.map(Miller(0,1,-1,-1,CSlist{2}),Miller(-1,1,0,-1,CSlist{2}),...
  Miller(1,0,-1,1,CSlist{2},'uvw'),Miller(1,0,-1,-1,CSlist{2},'uvw'));

istwinning = angle(gB_ZnZn(1).misorientation,twinning,'noSymmetry')
```
The output still gives one angle instead of six. The main goal of this question is to identify not only whether twinning is active, but also which of the six systems is active. So ideally we would define the variable "twinning"  for the six different twinning options in Zinc and apply each orientation.map in the "angle" function to be able to identify specific active twin system.

istwinning =

    2.1828

Thanks in advance.

G. Slokker

**Reply by s162555** [reply]
As a reaction to the previous note:

With the option "nosymmetry" it is possible to set the six different orientation.map for the six different twin options and for each one you can now calculate the angle right?

Thanks in advance

G. Slokker

**Comment by ralfHielscher** [comment]
I guess your 6 different twin options are
```matlab
twAxis = symmetrise(Miller(-2,1,1,0,CSlist{2}),'unique');
twinning = orientation.byAxisAngle(twAxis,60*degree);
```
You can use those in the angle command together with the option `noSym1` or `noSym2`.
You should very carefully think about what you are doing.

Ralf.
